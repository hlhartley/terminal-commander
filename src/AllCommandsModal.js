import React from 'react';
import './CSS/main.scss';

const AllCommandsModal = (props) => {
   let allCommands = props.terminalCommands;
   return (
   <div className='commands-modal'>
      <div className='button-container'>
         <button onClick={props.toggleModal}>x</button>
      </div>
      <h2 className='modal-title'><i class="fas fa-hand-point-right"></i>All Commands:</h2>
      <ul className='modal-list fa-ul'>
      {
         allCommands.map((command) => {
            return (
               <li key={command.id}><i class="fa-li fas fa-angle-double-right"></i><span className='skyblue'>{command.description}</span> = {command.command}</li>
            )
         })
      }
      </ul>
   </div>
   );
}

export default AllCommandsModal;