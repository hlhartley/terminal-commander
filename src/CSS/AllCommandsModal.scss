@import '_variables.scss';

.commands-modal {
    animation: popUp .5s ease 0s 1 normal;     
    background-color: $lightyellow;
    border: 3px solid black;
    border-radius: 15px;
    bottom: 0;
    height: 70%;
    left: 0;
    margin: auto;
    overflow: scroll;
    padding: 35px;
    position: absolute;
    right: 0;
    top: 30px;
    width: 65%;
    z-index: 10;
    button {
        background-color: $lightgrey;
        border: 2px solid black;
        border-radius: 50%;
    }
}
.button-container {
    align-items: right;
    display: flex;
    justify-content: flex-end;
}