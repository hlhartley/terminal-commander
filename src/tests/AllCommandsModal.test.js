import React from 'react';
import AllCommandsModal from '../AllCommandsModal';
import { shallow } from 'enzyme';

const hhTerminalCommands = [{
    id: 1,
    description: "Reset terminal screen",
    command: "reset"
    },
    {
    id: 2,
    description: "Clear screen",
    command: "clear"
}]

let wrapper;
describe('AllCommandsModal', () => {

  beforeEach(() => {
    wrapper = shallow(
        <AllCommandsModal 
            terminalCommands = {hhTerminalCommands} 
            toggleModal = { () => {} }
        />
    );
  })

  it('should exist', () => {  
    expect(wrapper).toBeDefined()
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})