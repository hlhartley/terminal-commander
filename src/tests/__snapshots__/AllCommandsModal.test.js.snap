// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllCommandsModal should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllCommandsModal
    terminalCommands={
      Array [
        Object {
          "command": "reset",
          "description": "Reset terminal screen",
          "id": 1,
        },
        Object {
          "command": "clear",
          "description": "Clear screen",
          "id": 2,
        },
      ]
    }
    toggleModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="button-container"
        >
          <button
            onClick={[Function]}
          >
            x
          </button>
        </div>,
        <h2
          className="modal-title"
        >
          <i
            class="fas fa-hand-point-right"
          />
          All Commands:
        </h2>,
        <ul
          className="modal-list fa-ul"
        >
          <li>
            <i
              class="fa-li fas fa-angle-double-right"
            />
            <span
              className="skyblue"
            >
              Reset terminal screen
            </span>
             = 
            reset
          </li>
          <li>
            <i
              class="fa-li fas fa-angle-double-right"
            />
            <span
              className="skyblue"
            >
              Clear screen
            </span>
             = 
            clear
          </li>
        </ul>,
      ],
      "className": "commands-modal",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
          >
            x
          </button>,
          "className": "button-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              class="fas fa-hand-point-right"
            />,
            "All Commands:",
          ],
          "className": "modal-title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "class": "fas fa-hand-point-right",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "All Commands:",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <i
                class="fa-li fas fa-angle-double-right"
              />
              <span
                className="skyblue"
              >
                Reset terminal screen
              </span>
               = 
              reset
            </li>,
            <li>
              <i
                class="fa-li fas fa-angle-double-right"
              />
              <span
                className="skyblue"
              >
                Clear screen
              </span>
               = 
              clear
            </li>,
          ],
          "className": "modal-list fa-ul",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  class="fa-li fas fa-angle-double-right"
                />,
                <span
                  className="skyblue"
                >
                  Reset terminal screen
                </span>,
                " = ",
                "reset",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "class": "fa-li fas fa-angle-double-right",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Reset terminal screen",
                  "className": "skyblue",
                },
                "ref": null,
                "rendered": "Reset terminal screen",
                "type": "span",
              },
              " = ",
              "reset",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  class="fa-li fas fa-angle-double-right"
                />,
                <span
                  className="skyblue"
                >
                  Clear screen
                </span>,
                " = ",
                "clear",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "class": "fa-li fas fa-angle-double-right",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Clear screen",
                  "className": "skyblue",
                },
                "ref": null,
                "rendered": "Clear screen",
                "type": "span",
              },
              " = ",
              "clear",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="button-container"
          >
            <button
              onClick={[Function]}
            >
              x
            </button>
          </div>,
          <h2
            className="modal-title"
          >
            <i
              class="fas fa-hand-point-right"
            />
            All Commands:
          </h2>,
          <ul
            className="modal-list fa-ul"
          >
            <li>
              <i
                class="fa-li fas fa-angle-double-right"
              />
              <span
                className="skyblue"
              >
                Reset terminal screen
              </span>
               = 
              reset
            </li>
            <li>
              <i
                class="fa-li fas fa-angle-double-right"
              />
              <span
                className="skyblue"
              >
                Clear screen
              </span>
               = 
              clear
            </li>
          </ul>,
        ],
        "className": "commands-modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
            >
              x
            </button>,
            "className": "button-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "x",
            "type": "button",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                class="fas fa-hand-point-right"
              />,
              "All Commands:",
            ],
            "className": "modal-title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "class": "fas fa-hand-point-right",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "All Commands:",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <i
                  class="fa-li fas fa-angle-double-right"
                />
                <span
                  className="skyblue"
                >
                  Reset terminal screen
                </span>
                 = 
                reset
              </li>,
              <li>
                <i
                  class="fa-li fas fa-angle-double-right"
                />
                <span
                  className="skyblue"
                >
                  Clear screen
                </span>
                 = 
                clear
              </li>,
            ],
            "className": "modal-list fa-ul",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    class="fa-li fas fa-angle-double-right"
                  />,
                  <span
                    className="skyblue"
                  >
                    Reset terminal screen
                  </span>,
                  " = ",
                  "reset",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "fa-li fas fa-angle-double-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Reset terminal screen",
                    "className": "skyblue",
                  },
                  "ref": null,
                  "rendered": "Reset terminal screen",
                  "type": "span",
                },
                " = ",
                "reset",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    class="fa-li fas fa-angle-double-right"
                  />,
                  <span
                    className="skyblue"
                  >
                    Clear screen
                  </span>,
                  " = ",
                  "clear",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "class": "fa-li fas fa-angle-double-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Clear screen",
                    "className": "skyblue",
                  },
                  "ref": null,
                  "rendered": "Clear screen",
                  "type": "span",
                },
                " = ",
                "clear",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
